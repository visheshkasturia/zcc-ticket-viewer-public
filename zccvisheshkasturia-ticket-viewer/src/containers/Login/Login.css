.Login {
  text-align: center;
  margin-top: 100px;
  margin-left: auto;
  margin-right: auto;
  height: 200px;
  width: 400px;
  border: solid 2px black;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.login-link{
  text-decoration: none;
  color: black;
  font-size: 18px;
  border: solid 1px black;
  border-radius: 8px;
  padding: 10px;
  background-color: #e7e7e7;
}
