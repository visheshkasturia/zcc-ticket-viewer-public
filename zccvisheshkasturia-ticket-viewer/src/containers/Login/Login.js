import React from 'react';
import './Login.css';

function Login() {
  return (
    <div className="Login">
      <h2 data-testid="h2-login-test" className="h2-login">Zendesk Ticket Viewer</h2>
      <a data-testid="a-login-test" className="login-link" href="http://localhost:5000/oauth"> Login with Zendesk</a>
    </div>
  );
}

export default Login;
