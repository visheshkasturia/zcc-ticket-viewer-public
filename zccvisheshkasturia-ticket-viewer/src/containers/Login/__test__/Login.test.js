import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Login from '../Login';

afterEach(cleanup);
describe('Login renders correctly', () => {
  it('<h2> has correct text', () => {
    const { getByTestId } = render(<Login />);
    expect(getByTestId('h2-login-test')).toHaveTextContent('Zendesk Ticket Viewer');
  });

  it('Link has correct text', () => {
    const { getByTestId } = render(<Login />);
    expect(getByTestId('a-login-test')).toHaveTextContent('Login with Zendesk');
  });

  it('Link has correct attibute', () => {
    const { getByTestId } = render(<Login />);
    expect(getByTestId('a-login-test')).toHaveAttribute('href', 'http://localhost:5000/oauth');
  });

  it('Matches snapshot', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
