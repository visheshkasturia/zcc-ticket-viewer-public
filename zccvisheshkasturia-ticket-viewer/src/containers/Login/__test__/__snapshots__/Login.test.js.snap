// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly Matches snapshot 1`] = `
<div
  className="Login"
>
  <h2
    className="h2-login"
    data-testid="h2-login-test"
  >
    Zendesk Ticket Viewer
  </h2>
  <a
    className="login-link"
    data-testid="a-login-test"
    href="http://localhost:5000/oauth"
  >
     Login with Zendesk
  </a>
</div>
`;
